backend/app/quiz/schemas.py

```
@@ -0,0 +1,105 @@
+
+
+class QuizGenerateRequest(BaseModel):
+    """Schema for generating quiz questions."""
+    topic_id: int = Field(..., description="Topic to generate questions for")
+    count: int = Field(5, ge=1, le=20, description="Number of questions")
+    difficulty: Optional[str] = Field(None, description="Filter by difficulty")
+    
+    @validator('difficulty')
+    def validate_difficulty(cls, v):
+        if v and v not in ['easy', 'medium', 'hard']:
+            raise ValueError('Difficulty must be easy, medium, or hard')
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)